### PR TITLE
Building a docker container to run our builds in

### PR DESCRIPTION
**Pull request by mdbenjam**
Resolves https://github.com/department-of-veterans-affairs/caseflow/issues/5062

### Description
This PR converts the prereqs script into steps in our own circleci docker container. We cut out a lot the steps by picking a circleci docker container that already has yarn/node installed. We also add a step to install intantclient to allow connecting to FACOLS.

### Acceptance Criteria 
- [ ] Builds still pass
- [ ] All gems including ruby-oci install properly

### Testing Plan
1. CI passes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
 
       - run:
           name: bundle install
-          command: bundle check || bundle install --deployment --without production staging ssh_forwarding local
```
Comment (sansb) [inline .circleci/config.yml]: Curious why we don't need to specify the groups here

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ jobs:
       # The first image listed is the image that all steps run inside of.
       # This can be modified by steps, or can be built as a separate
       # customized container.
-      - image: circleci/ruby:2.2-browsers
+      - image: 008577686731.dkr.ecr.us-gov-west-1.amazonaws.com/circleci:latest
```
Comment (sunil-sadasivan) [inline .circleci/config.yml]: awesome!

Comment (sunil-sadasivan) [reply .circleci/config.yml]: curious, did you have to specify AWS creds on circle CI to get this to work with ECR?

File: ci-bin/circle_docker_container/build_and_push.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#! /bin/sh
```
Comment (sunil-sadasivan) [inline ci-bin/circle_docker_container/build_and_push.sh]: 👍

File: ci-bin/circle_docker_container/build_and_push.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  aws s3 cp s3://shared-s3/dsva-appeals/instant-client-12-1.tar.gz instant-client-12-1.tar.gz
+  tar xvzf instant-client-12-1.tar.gz
+fi
+
+rm instant-client-12-1.tar.gz
+
+eval $(aws ecr get-login --no-include-email --region us-gov-west-1)
+
+docker build -t circleci .
+docker tag circleci:latest 008577686731.dkr.ecr.us-gov-west-1.amazonaws.com/circleci:latest
```
Comment (sunil-sadasivan) [inline ci-bin/circle_docker_container/build_and_push.sh]: curious, is it ideal to keep pushing builds with the `latest` tag, or should we add versions?  No major preference myself, was just wondering if you thought of that.  I'd imagine that makes it a bit of a hassle as we'd have to specify the build version circleci picks up.

File: .circleci/config.yml (hunk 2 above)
Comment (mdbenjam) [reply .circleci/config.yml]: Yup, @lisac-usds and @RobertAlexanderUSDS are awesome and set us up with a circleCI user and role that only has access to this repository.

File: ci-bin/circle_docker_container/build_and_push.sh (hunk 4 above)
Comment (mdbenjam) [reply ci-bin/circle_docker_container/build_and_push.sh]: No I hadn't, what would be the advantage of that?

Comment (nemodjuric) [reply ci-bin/circle_docker_container/build_and_push.sh]: "CircleCI cannot guarantee that mutable tags will return an up-to-date version of an image. You could specify alpine:latest and actually get a stale cache from a month ago."(taken from https://circleci.com/docs/2.0/executor-types/#docker-image-best-practices) We should consider putting a non-mutable tag to our images in the future.

**Pull request merged**